src/Build/Graph/ProjectGraph.cs

```diff
@@ -396,7 +415,9 @@ public ProjectGraph(ProjectGraphEntryPoint entryPoint, ProjectCollection project
         {
             ErrorUtilities.VerifyThrowArgumentNull(projectCollection, nameof(projectCollection));
 
-            projectInstanceFactory = projectInstanceFactory ?? DefaultProjectInstanceFactory;
+            var timer = Stopwatch.StartNew();
```

Comment:
Should the timer be here in debug mode only?

Reply:
The intent is to always log graph construction metrics.

Reply:
Can you please log Start/Stop ETW events as well? (ref: src/Framework/MSBuildEventSource.cs)

This will be super useful for automated perf runs to make sure that the graph building and evaluation performance hasn't regressed.

Reply:
You could theoretically get this from grepping through a log, right? It's just messier As far as I'm concerned, either approach is fine, and ETW events would be lighter, but this form of logging it would run every time, whereas ETW events only fire when they're enabled.

Reply:
Agree with @ladipro: also add ETW.